framework/include/libtorch/userobjects/TorchScriptUserObject.h

```diff
@@ -0,0 +1,53 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef LIBTORCH_ENABLED
+
+#pragma once
+
+// MOOSE includes
+#include "GeneralUserObject.h"
+#include "TorchScriptModule.h"
+
+/**
+ * A user object the loads a torch module using the
+ * torch script format and just-in-time compilation.
+ */
+class TorchScriptUserObject : public GeneralUserObject
+{
+public:
+  static InputParameters validParams();
+
+  TorchScriptUserObject(const InputParameters & parameters);
+
+  virtual void initialize() override{};
+  virtual void execute() override;
+  virtual void finalize() override{};
+
+  /**
+   * Get access to the torch script module which was loaded in this object.
+   */
+  const std::shared_ptr<Moose::TorchScriptModule> & module() const { return _torchscript_module; }
+
+  /**
+   * Function to evaluate the torch script module at certain input.
+   * @param input The input tensor. Unfortunately, this cannot be const since it creates a graph in
+   * the background.
+   */
+  torch::Tensor evaluate(torch::Tensor & input) const;
```

Comment:
Why not `const torch::Tensor & input`?

Reply:
The intermediate tensors you create during forward evaluation based on the input tensor will carry over the graph. That's not a problem I think. You could easily verify this.

Reply:
Will look into it!

Reply:
Nice, you were right! Will push the fix in the next commit.